src/MediaStreamTrack.js

```diff
@@ -0,0 +1,118 @@
+
+import { NativeModules } from 'react-native';
+import { defineCustomEventTarget } from 'event-target-shim';
+
+import { deepClone } from './RTCUtil';
+
+const { WebRTCModule } = NativeModules;
+
+const MEDIA_STREAM_TRACK_EVENTS = ['ended', 'mute', 'unmute'];
+
+type MediaStreamTrackState = 'live' | 'ended';
+
+class MediaStreamTrack extends defineCustomEventTarget(...MEDIA_STREAM_TRACK_EVENTS) {
+    _constraints: Object;
+    _enabled: boolean;
+    _settings: Object;
+
+    id: string;
+    kind: string;
+    label: string;
+    muted: boolean;
+    // readyState in java: INITIALIZING, LIVE, ENDED, FAILED
+    readyState: MediaStreamTrackState;
+    remote: boolean;
+
+    constructor(info) {
+        super();
+
+        this._constraints = info.constraints || {};
+        this._enabled = info.enabled;
+        this._settings = info.settings || {};
+
+        this.id = info.id;
+        this.kind = info.kind;
+        this.label = info.label;
+        this.muted = false;
+        this.remote = info.remote;
+
+        const _readyState = info.readyState.toLowerCase();
+        this.readyState = _readyState === 'initializing' || _readyState === 'live' ? 'live' : 'ended';
+    }
+
+    get enabled(): boolean {
+        return this._enabled;
+    }
+
+    set enabled(enabled: boolean): void {
+        if (enabled === this._enabled) {
+            return;
+        }
+        WebRTCModule.mediaStreamTrackSetEnabled(this.id, !this._enabled);
+        this._enabled = !this._enabled;
+        this.muted = !this._enabled;
+    }
+
+    stop() {
+        WebRTCModule.mediaStreamTrackSetEnabled(this.id, false);
+        this.readyState = 'ended';
+        // TODO: save some stopped flag?
+    }
+
+    /**
+     * Private / custom API for switching the cameras on the fly, without the
+     * need for adding / removing tracks or doing any SDP renegotiation.
+     *
+     * This is how the reference application (AppRTCMobile) implements camera
+     * switching.
+     */
+    _switchCamera() {
+        if (this.remote) {
+            throw new Error('Not implemented for remote tracks');
+        }
+        if (this.kind !== 'video') {
+            throw new Error('Only implemented for video tracks');
+        }
+        return WebRTCModule.mediaStreamTrackSwitchCamera(this.id);
```

Comment:
i'm out of my element here but... in the merge conflict resolution it looks like this changes react's version which does not return here. did you mean to put the `return` back in?

Reply:
One of the changes that we have made at our version of react-native-webrtc, is to return when we are switching the camera, which is the new facingMode result. That is why we have this return.

Reply:
👍  just wanted to make sure it was intentional.